### PR TITLE
Add Dependabot guide

### PR DESCRIPTION
**Pull request by joshspicer**
Prepping documentation update for release of dependabot integration of dev containers 🥳 

**Note that this support is not GA yet**.  I will merge this doc once the functionality has landed for general use!  Please keep an eye out here for the "OK" to add these workflows to your repos!

### CODE REVIEW

File: _posts/2024-01-23-dependabot.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+ ```yaml
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "devcontainers" # See documentation for possible values
```
Comment (samruddhikhandale) [inline _posts/2024-01-23-dependabot.md:23]: Unrelated question: I was trying out this change and can't help but notice the red squiggles. Is there a way to make it go away? 🤔 

image

File: _posts/2024-01-23-dependabot.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ {
+     "image": "mcr.microsoft.com/devcontainers/base:jammy",
+     "features": {
+-        "ghcr.io/devcontainers/features/docker-in-docker:1": {}
++        "ghcr.io/devcontainers/features/docker-in-docker:2": {}
+     }
+ }
+```
+
+ This updater ensures publicly-accessible Features are pinned to the latest version in the associated `devcontainer.json` file.  If a dev container has an associated lockfile, that file will also be updated. For more information on lockfiles, see this [specification](https://github.com/devcontainers/spec/blob/main/docs/specs/devcontainer-lockfile.md).
```
Comment (samruddhikhandale) [inline _posts/2024-01-23-dependabot.md:77]:
```suggestion
 This update ensures publicly-accessible Features are pinned to the latest version in the associated `devcontainer.json` file.  If a dev container has an associated lockfile, that file will also be updated. For more information on lockfiles, see this [specification](https://github.com/devcontainers/spec/blob/main/docs/specs/devcontainer-lockfile.md).
```

**Review by samruddhikhandale — APPROVED** [review]
⚡ ⚡  Super cool, yay!

File: _posts/2024-01-23-dependabot.md (hunk 1 above)
Comment (samruddhikhandale) [reply _posts/2024-01-23-dependabot.md:23]: Would be cool to see `devcontainers` in the suggestions 🤩 

image

**Review by bamurtaugh — APPROVED** [review]
So excited to see this, and thanks so much for writing this up too!

File: _posts/2024-01-23-dependabot.md (hunk 1 above)
Comment (joshspicer) [reply _posts/2024-01-23-dependabot.md:23]: The functionality is still behind a flag, so don't expect it to work yet :)

We've updated the schema here https://github.com/SchemaStore/schemastore/pull/3543.  Circling back with the dependabot folks to see if there's anything else we need to do

Comment (samruddhikhandale) [reply _posts/2024-01-23-dependabot.md:23]: Uff, gotcha! Thanks!

File: _posts/2024-01-23-dependabot.md (hunk 2 above)
Comment (joshspicer) [reply _posts/2024-01-23-dependabot.md:77]: That was intentional, using "updater" here shortening "dependabot version updater"

**Comment by eitsupi** [comment]
- Is this also supported for files with a pattern of `./.devcontainer/**/devcontainer.json`? (When I looked at the PR, I felt that it might not be supported). If it's not supported by default, I think it's worth mentioning that the directory configuration can cover it.
- Is this support the image tag updatting e.g. `"image": "ubuntu:20.04"`? If dependabot don't support this yet, users may be looking for an alternative using a Dockerfile e.g. `"build": { "dockerfile": "Dockerfile" }`

**Comment by joshspicer** [comment]
Hey @eitsupi!

- Yes, that pattern is supported (eg: `.devcontainer/foo/devcontainer.json`) without needing to change the `directory` variable.  You are correct that early iterations of the dependabot-core PR did not evaluate all possible paths, but it does now :) 
- Nope, this first release just looks at Features.  I'm hoping to expand on this ecosystem in the future though to do just that 🤞

**Comment by eitsupi** [comment]
Awesome, thanks for the great works!

**Review by chrmarti — APPROVED** [review]
Great work! Is there an option to have it update the lockfile, but not the versions in the devcontainer.json, so a user could, e.g, stay on major version 1 and get the latest version for that?

**Comment by joshspicer** [comment]
@chrmarti Not currently, both the lockfile and devcontainer.json will be upgraded in tandem.  

The typical dependabot commands are supported like `@dependabot ignore this major version`, so a user could ignore all updates of a particular major version, but in that case it also won't suggest lockfile updates.  That sounds like a good lever to add!

**Pull request merged**